docs/en/concepts-and-designs/lal.md

```diff
@@ -0,0 +1,191 @@
+# Log Analysis Language
+
+Log Analysis Language (LAL) in SkyWalking is essentially a Domain-Specific Language (DSL) to analyze logs. You can use
+LAL to parse, extract, filter, analyze, and save the logs, as well as collaborate the logs with traces, and metrics.
+
+Check the [`default.yaml`](../../../oap-server/server-bootstrap/src/main/resources/lal/default.yaml) for examples.
+
+## Filter
+
+A filter is a group of [parser](#parser), [extractor](#extractor) and [sink](#sink). Users can use one or more filters
+to organize their processing logics. Every piece of log will be sent to all filters in an LAL rule. The piece of log sent into the filter is available as property `log` in the LAL, therefore you can access the log level via `log.level`.
+
+### Parser
+
+Parsers are responsible for parsing the raw logs into structured data in SkyWalking for further processing. There are 3
+types of parsers at the moment, namely `json`, `yaml`, and `text`.
+
+When a piece of log is parsed, there is a corresponding property available, called `parsed`, injected by LAL.
+Property `parsed` is typically a map, containing all the fields parsed from the raw logs, for example, if the parser
+is `json` / `yaml`, `parsed` is a map containing all the key-values in the `json` / `yaml`, if the parser is `text`
```

Comment:
Speaking of the JSON and YAML, are they going to be flat structure? Such as using `level1.level2...` to represent the hierarchy?

Reply:
> Speaking of the JSON and YAML, are they going to be flat structure? Such as using `level1.level2...` to represent the hierarchy?

It doesn't make things different in terms of usages, in LAL, they are nested objects under the hood, but you can still use `a.b.c` to access the nested properties.